BinaryEncoding.md

```diff
@@ -3,120 +3,143 @@
+  - ```uint8```: The number of parameters
+  - ```uint8```: The function return type, as a [LocalType](https://github.com/v8/v8/blob/master/src/wasm/wasm-opcodes.h#L16)
+  - For each parameter:
+    + ```uint8```: The parameter type, as a LocalType
+
+### Functions section
+This section must be preceded by a [Signatures](#signatures-section) section.
+
+* ```varuint32```: The number of functions in the section
+* For each function:
+  - ```uint8```: The [function declaration bits](https://github.com/v8/v8/blob/master/src/wasm/wasm-module.h#L39)
+  - ```uint16```: The function signature (as an index into the Signatures section)
+  - If the ```kDeclFunctionName``` bit is set:
+    + ```uint32```: The offset of the function name in the file.
+  - If the ```kDeclFunctionImport``` bit is set, **the function entry ends here**
+  - If the ```kDeclFunctionLocals``` bit is set:
+    + ```uint16```: The number of i32 locals
+    + ```uint16```: The number of i64 locals
+    + ```uint16```: The number of f32 locals
+    + ```uint16```: The number of f64 locals
+  - ```uint16```: The size of the function body, in bytes
+  - The function body
+
+### Globals section
+* ```varuint32```: The number of global variable declarations in the section.
+* For each global variable:
+  - ```uint32```: The offset of the global variable name in the file.
+  - ```uint8```: The type of the global, as a [MemType](https://github.com/v8/v8/blob/master/src/wasm/wasm-opcodes.h#L25)
+  - ```uint8```: ```1``` if the global is exported
+
+### Data Segments section
+* ```varuint32```: The number of data segments in the section.
+* For each data segment:
+  - ```uint32```: The base address of the data segment in memory.
+  - ```uint32```: The offset of the data segment's data in the file.
+  - ```uint32```: The size of the data segment (in bytes)
+  - ```uint8```: ```1``` if the segment's data should be automatically loaded into memory at module load time.
+
+### Function Table section
+This section must be preceded by a [Functions](#functions-section) section.
+
+* ```varuint32```: The number of function table entries in the section
+* For each function table entry:
+  - ```uint16```: The index of the function (in the [Functions](#functions-section) section's index space)
+
+### WLL section
+
+* ```varuint32```: The size of the section body, in bytes
+* The section body (contents currently undefined)
+
+### End section
+This indicates the end of the module's sections. Additional data can follow this section marker (for example, to store function names or data segment bodies) but it is not explicitly handled by the decoder.
+
+# v8-native prototype format
+
+The native prototype built for [V8](https://github.com/v8/v8/blob/master/src/wasm)
+implements a binary format that embodies many of the ideas described in this document.
+It is described in detail in a [public design doc](https://docs.google.com/document/d/1-G11CnMA0My20KI9D7dBR6ZCPOBCRD0oCH6SHCPFGx0/edit?usp=sharing).
```

Comment:
You could word this as "This document is based on but supersedes the format developed for V8 and documented here: [link]"